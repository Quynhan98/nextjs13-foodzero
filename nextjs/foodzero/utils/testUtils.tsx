import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { BookingProvider } from '@contexts/BookingProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <AuthProvider>
        <BookingProvider>{component}</BookingProvider>
      </AuthProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
