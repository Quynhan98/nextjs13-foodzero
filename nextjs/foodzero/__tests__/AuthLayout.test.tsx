// Utils
import { render } from '@utils/testUtils'

// App
import AuthLayout from 'app/(auth)/layout'

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}))

describe('Auth Layout render', () => {
  it('Should Auth Layout match Snapshot', () => {
    const { container } = render(<AuthLayout>Contents Page</AuthLayout>)

    expect(container).toMatchSnapshot()
  })
})
