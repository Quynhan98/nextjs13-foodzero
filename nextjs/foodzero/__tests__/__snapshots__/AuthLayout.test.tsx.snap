// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Layout render Should Auth Layout match Snapshot 1`] = `
<div>
  <html
    lang="en"
  >
    <body>
      <noscript>
        &lt;iframe src="https://www.googletagmanager.com/ns.html?id=GTM-NVSQVJS"
            height="0" width="0" style="display:none;visibility:hidden"&gt;&lt;/iframe&gt;
      </noscript>
      <main
        class="css-lepva3"
      >
        Contents Page
      </main>
      <span
        hidden=""
        id="__chakra_env"
      />
    </body>
  </html>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
